pkg/reconciler/service/workerpool.go

```diff
@@ -53,23 +74,86 @@ func (pb *workPoolBuilder) Build(ctx context.Context) (*WorkerPool, error) {
 		return nil, err
 	}
 	pb.workerPool.antsPool = antsPool
+	pb.workerPool.PoolID = uuid.NewString()
 
 	go func(ctx context.Context, antsPool *ants.Pool) {
-		<-ctx.Done()
-		log.Info("Shutting down worker pool")
-		antsPool.Release()
+		ticker := time.NewTicker(defaultInterval)
```

Comment:
From architectural side, we should think to move this logic into a separte module of the component-reconcilers as the measuring of the metrics (currently, we measure only the worker pool, but I assume a few more values will be added) should not be connected ot the worker-pool module itself.